sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -86,12 +96,20 @@ def _predict_in_or_out_of_sample(
         level_arguments = None if levels is None else [100 * level for level in levels]
 
         if fh_type == "in-sample":
-            predictions = self._forecaster.predict_in_sample(level=level_arguments)
+            predict_method = self._forecaster.predict_in_sample
```

Comment:
I believe you also raised the point that this method itself may be absent for older versions? We can handle it as either a specific lower bound, or via another tag check. As earlier, my first vote will still be to use `1.5.0` as lower bound as that's when the generalised class and new models were supported, and I'm reasonably optimistic that Nixtla will ensure this in future.

But @fkiraly please share your opinion as well.

Reply:
hm, I usually err on the side of greater compatibility if it is not too much of a hassle.

How about: we have a checker function as suggested, `_supports_proba`, which returns two booleans. At the very top, we have a `hasattr` check for the method?